### PR TITLE
Update express & rewrite main.js to es6

### PR DESCRIPTION
**Pull request by AntonPuko**
1) clean as much depreciation errors as I could find. there is  a big chance that there many more errors, will fix them when find.
2) rewrite main.js in es6 style.

seems  "node-oauth2-server"  doen't works right in the current version of server, and checks nothing.. that way we can do whatever we want from particle-cli with any auth_token provided.

### CODE REVIEW

**Comment by jlkalberer** [comment]
The other guys used an express oauth package that wrapped the node oauth one.

**Comment by AntonPuko** [comment]
ye, and they did many changes related with that I'll take that task next.

File: src/main.js [hunk 1]
```diff
@@ -16,102 +16,100 @@
...
+import UserCreator from './lib/UserCreator.js';
+
+import api from './views/api_v1.js';
+import eventsV1 from './views/EventViews001.js';
+import Webhook from './views/Webhooks';
+
+const  NODE_PORT = process.env.NODE_PORT || 8080;
+
+// TODO wny do we need this? (Anton Puko)
+global._socket_counter = 1;
```
Comment (jlkalberer) [inline src/main.js]: I want to get rid of all the `global` calls.  It is really shitty practice.

I think what this is for is to keep track of each socket that gets opened in order to build an id or something

File: src/main.js [hunk 2]
```diff
@@ -16,102 +16,100 @@
...
-var server = new DeviceServer({
-	coreKeysDir: settings.coreKeysDir
+const deviceServer = new DeviceServer({
+	coreKeysDir: settings.coreKeysDir,
 });
-global.server = server;
-server.start();
+
+// TODO wny do we need next line? (Anton Puko)
+global.server = deviceServer;
```
Comment (jlkalberer) [inline src/main.js]: Again.. this is stupid globals :(

**Pull request merged**